cli/silence_import.go

```diff
@@ -0,0 +1,67 @@
+package cli
+
+import (
+	"os"
+	"io/ioutil"
+	"encoding/json"
+	"fmt"
+
+	"github.com/prometheus/alertmanager/types"
+	"github.com/spf13/cobra"
+	"github.com/pkg/errors"
+)
+
+var importCmd = &cobra.Command{
+	Use:   "import [JSON file]",
+	Short: "Import silences",
+	Long: `Import alertmanager silences from JSON file or stdin
+
+  This command can be used to bulk import silences from a JSON file
+  created by query command. For example:
+
+  amtool silence query -o json foo > foo.json
+  amtool silence import foo.json
+
+  JSON data can also come from stdin if no param is specified.
+	`,
+	Args: cobra.MaximumNArgs(1),
+	Run:  CommandWrapper(bulkImport),
+}
+
+func bulkImport(cmd *cobra.Command, args []string) error {
+	input := os.Stdin
+	var err error
+
+	if len(args) == 1 {
+		input, err = os.Open(args[0])
+
+		if err != nil {
+			return err
+		}
+	}
+
+	data, err := ioutil.ReadAll(input)
+	if err != nil {
+		return err
+	}
+
+	var silences []types.Silence
+	err = json.Unmarshal(data, &silences)
+	if err != nil {
+		return errors.Wrap(err, "couldn't unmarshal input data, is it in JSON?")
+	}
+
+	for _, silence := range silences {
+		// reset the ID, otherwise alertmanager API will try to replace an existing silence
+		// this *might* create duplicated silences but that should generally be OK
+		silence.ID = ""
+
+		err = addSilence(silence)
```

Comment:
Since this is being done in serial, it will be relatively slow. We can always create a small worker pool (https://gobyexample.com/worker-pools) without much trouble. I'm fine adding it later, but if you want to add it now, feel free.